protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -800,7 +806,12 @@ func (rpccs *RPCConsumerServer) relayInner(ctx context.Context, singleConsumerSe
 		metadataAdd := metadata.New(map[string]string{common.IP_FORWARDING_HEADER_NAME: consumerToken})
 		connectCtx = metadata.NewOutgoingContext(connectCtx, metadataAdd)
 		defer connectCtxCancel()
+
+		// add consumer processing timestamp before provider metric and start measuring time after the provider replied
+		rpccs.rpcConsumerLogs.AddMetricForProcessingLatencyBeforeProvider(analytics, rpccs.listenEndpoint.ChainID, rpccs.listenEndpoint.ApiInterface)
+
 		reply, err = endpointClient.Relay(connectCtx, relayRequest, grpc.Trailer(&relayResult.ProviderTrailer))
+		analytics.SetProcessingTimestamp(time.Now())
```

Comment:
setting timestamp before error validation can cause miscalculations on retry attempts. 

set timestamp only after error parsing and check if value is set in the return calculation. if not, its a retry flow therefore not applicable.

Reply:
done. see https://github.com/lavanet/lava/commit/f0dc3a7e3b89fd1eaf6fec5a3ba915be041519f8

Reply:
see new fix: https://github.com/lavanet/lava/commit/3563b98eaf047490c45593deeb13d30c8e863def

Reply:
removed redundant function here: https://github.com/lavanet/lava/commit/9a956dc33c8fff4adeb7d4c5b88e67317175d431